site-src/geps/gep-1709.md

```diff
@@ -37,8 +45,295 @@ which implementations can prove they satisfy and be recognized for.
+> to fulfill conformance tests for multiple profiles (e.g. `Layer4Core` and
+> `Layer7Extended`, or perhaps an implementation can literally do everything,
+> including `Mesh`!). Rather than bother with even higher level compilations at
+> this point, we will simply report these separately and the implementations
+> can indicate that they support multiples to their end-users in whatever way
+> best suits them. If over time we find that the community is really strongly
+> in favor of compilations at a higher level we are set up to add them in a
+> follow up iteration.
+
+[camelcase]:https://en.wikipedia.org/wiki/Camel_case
+[feat]:https://github.com/kubernetes-sigs/gateway-api/blob/c61097edaa3b1fad29721e787fee4b02c35e3103/conformance/utils/suite/suite.go#L33
+
+### Integration
+
+Integrating the test suite into your implementation can be done using one of
+the following methods:
+
+- The [go test][go-test] command line interface which enables projects of any
+  language to run the test suite on any platform [Golang][go] supports.
+- Using the conformance test suite as a [Golang library][lib] within an already
+  existing test suite.
+
+> **NOTE**: Usage as a library is already an established colloquialism in the
+> community, this effort simply intends to make that more official.
+
+Selected conformance profiles can be identified as arguments in either
+configuration. For instance using the command line interface method to run the
+`Layer7Core` profile to verify core feature conformance:
+
+```console
+$ go test ./conformance/... -args -gateway-class=acme -conformance-profile=Layer7Core
+```
+
+Or the equivalent configuration using the Golang library:
+
+```go
+cSuite, err := suite.New(suite.Options{
+    GatewayClassName: "acme",
+    Profile: Layer7CoreProfile,
+    // other options
+})
+require.NoError(t, err, "misconfigured conformance test suite")
+cSuite.Setup(t)
+
+for i := 0; i < len(tests.ConformanceTests); i++ {
+    test := tests.ConformanceTests[i]
+    test.Run(t, cSuite)
+}
+```
+
+> **NOTE**: In the `suite.Options` above it's still possible to add `SkipTests`
+> but when used in conjunction with `Profile` this will result in a report that
+> the profile is not valid for reporting. Implementations in this state may be
+> able to report themselves as "in progress", see the
+> [certification section](#certification) for details.
+
+Alternatively for an `Extended` conformance profile where not all of the
+features are implemented (as described in the [profiles](#profiles) section
+above):
+
+```console
+$ go test ./conformance/... -args \
+    -gateway-class=acme \
+    -conformance-profile=Layer7Extended \
+    -unsupported-features=HTTPResponseHeaderModification,HTTPRouteMethodMatching,HTTPRouteQueryParamMatching,
+```
+
+Or the equivalent configuration using the Golang library:
+
+```go
+cSuite, err := suite.New(suite.Options{
+    GatewayClassName: "acme",
+    Profile: Layer7ExtendedProfile,
+    UnsupportedFeatures: sets.New(
```

Comment:
Feels like we can combine `UnsupportedFeatures` and `SkipTests` to one field and have additional logic that checks whether this combined field is used with `Core` profile to check validity.

Reply:
I think `UnsupportedFeatures` and `SkipTests` need to be kept separate as `SkipTests` is particularly helpful _during_ the development process. E.g. someone could have partial support for a _feature_ but not all tests passing just yet and use `SkipTests` to disable the tests they're unable to fulfill just yet while they otherwise plan the work to fix it. We did literally this downstream at Kong and it was pretty convenient.

I'll consider this one resolved, but if you still feel there's more to do here please don't hesitate to unresolve and let me know if I've missed something or there's something else you feel needs changing.